### PR TITLE
update `rm -rf /opt/hostedtoolcache` avoid change the python version

### LINKED ISSUE #7416: `rm -rf /opt/hostedtoolcache` may change the python version actually used to run CI tests
**Describe the bug**
When I run some tests on github action in model zoo repo, I found that the following line in the action yml file will remove the installed python version:
```
rm -rf /opt/hostedtoolcache
```
It may lead to the system's default python version will be used to run the tests.

@KumoLiu and I simply check MONAI repo, and found that at least the `deploy` action is impacted. In the corresponding yml file:
https://github.com/Project-MONAI/MONAI/blob/6c9e49ee045c345ecd8d3518cb0d503ed02c61b7/.github/workflows/setupapp.yml#L103
`rm -rf /opt/hostedtoolcache` is used after python env is set. 

If we look at the latest deploy action job, we can see that the python3.9 test actually uses python 3.10 which is not expected.

https://github.com/Project-MONAI/MONAI/actions/runs/7648726812/job/20842001933#step:7:411

Hi @wyli @KumoLiu , I think we should think about how to install the expected python version back before running tests if need `rm -rf /opt/hostedtoolcache`

### PR DESCRIPTION
**Pull request by KumoLiu**
Fixes #7416

### Description

update `rm -rf /opt/hostedtoolcache` avoid change the python version

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by ericspod** [comment]
Why is this directory being deleted in the original? I feel it would be helpful to add a comment at the first instance of this pattern explaining the rationale so if we come back to it like we're doing now we have a better idea of what's going on.

**Comment by KumoLiu** [comment]
> Why is this directory being deleted in the original? I feel it would be helpful to add a comment at the first instance of this pattern explaining the rationale so if we come back to it like we're doing now we have a better idea of what's going on.

Hi Eric, the original reason is the resource issue from github action.
https://github.com/Project-MONAI/MONAI/issues/7030

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -62,7 +62,7 @@ jobs:
     - name: docker_build
       shell: bash
       run: |
-        rm -rf /opt/hostedtoolcache
+        find /opt/hostedtoolcache/* -maxdepth 0 ! -name 'Python' -exec rm -rf {} \;
```
[the remark this answers is not in the document]
Comment (KumoLiu) [reply .github/workflows/docker.yml:68]: Related job listed here.
https://github.com/Project-MONAI/MONAI/actions/runs/7709222397/job/21010019425
https://github.com/Project-MONAI/MONAI/actions/runs/7709222387/job/21010294756#step:4:1
https://github.com/Project-MONAI/MONAI/actions/runs/7709222388/job/21009953825#step:6:14

**Comment by KumoLiu** [comment]
Hi @ericspod, @yiheng-wang-nv, could you please help review it again, thanks.

**Review by ericspod — APPROVED** [review]
Looks good to me.

**Pull request merged**